src/np_codeocean/upload.py

```diff
@@ -83,6 +83,35 @@ def create_ephys_symlinks(session: np_session.Session, dest: Path,
         if not abs_path.is_dir():
             np_tools.symlink(as_posix(abs_path), dest / rel_path)
     logger.debug(f'Finished creating symlinks to raw ephys data files in {root_path}')
+    correct_structure(dest)
+
+def correct_structure(dest: Path) -> None:
+    """
+    In case some probes are missing, remove device entries from structure.oebin files with folders that don't actually exist.
+    """
+    logger.debug('Creating modified structure.oebin')
+    oebin_paths = dest.glob('**/*structure.oebin')
+    for oebin_path in oebin_paths:
+        logger.debug(f'Examining oebin: {oebin_path} for correction')
+        oebin_obj = np_tools.read_oebin(np_config.normalize_path(oebin_path.readlink()))
+
+        events_dir = oebin_path.parent / 'events'
+        # iterate over copy of list so as to not disrupt iteration when elements are removed
+        for device in [device for device in oebin_obj['events']]:
+            if list(events_dir.rglob(device['folder_name'])) == []:
```

Comment:
just to make it clear that this isn't a hunting expedition - we want to check precisely one thing:
```python 
if not (events_dir / device['folder_name']).exists():
   ...
```